tasks/data_handling/data_pre_processing.py

```diff
@@ -0,0 +1,149 @@
+import scanpy as sc
+
+def create_count_matrix(file_path : str = None) -> sc.AnnData:
+    """
+    ### Function that transforms h5-file to a count matrix.
+
+    ---
+    
+    #### Args:
+        - file_path (str): Path to a 10x hdf5 file.
+
+    #### Returns: 
+        - AnnData: Count matrix from h5-file in file_path with unique variables
+
+    ---
+
+    Annotated data matrix, where observations/cells are named by their barcode and variables/genes by gene name. Stores the following information:
+
+    `~anndata.AnnData.X`
+        The data matrix is stored
+    `~anndata.AnnData.obs_names`
+        Cell names
+    `~anndata.AnnData.var_names`
+        Gene names
+    `~anndata.AnnData.var`\ `['gene_ids']`
+        Gene IDs
+    `~anndata.AnnData.var`\ `['feature_types']`
+        Feature types
+    ---
+
+    Written: ronjah@chalmers.se
+    """
+    if file_path == None:
+        raise ValueError('Need a path to h5-file.')
+    adata = sc.read_10x_h5(file_path)
+    adata.var_names_make_unique()
+    return adata
+
+
+def quality_control(adata: sc.AnnData = None) -> None:
+    """
+    ### Function that adds QC to AnnData object
+
+    ---
+    
+    #### Args:
+        - adata (AnnData): AnnData object to perform quality control on.
+
+    #### Returns: 
+        - None
+
+    ---
+    Written: ronjah@chalmers.se
+    """
+    adata.var['mt'] = adata.var_names.str.startswith('MT-')  # annotate the group of mitochondrial genes as 'mt'
+    sc.pp.calculate_qc_metrics(adata, qc_vars=['mt'], percent_top=None, log1p=False, inplace=True)
+
+
+def remove_bad_cells(adata: sc.AnnData = None) -> sc.AnnData:
+    """
+    ### Function that removes bad cells
+
+    ---
+    
+    #### Args:
+        - adata (AnnData): AnnData object that has gone through QC.
+
+    #### Returns: 
+        - AnnData: Where bad cells has been removed.
+
+    ---
+    Written: ronjah@chalmers.se
+    """
+    adata = adata[adata.obs.n_genes_by_counts < 2500, :]
+    adata = adata[adata.obs.n_genes_by_counts > 200, :]
+    adata = adata[adata.obs.pct_counts_mt < 5, :]
+    return adata
+
+# 3: Normalizing the data. LogNormalize?
```

Comment:
Refactor to standardized comments. If this is for you to return and work back on this, use the conventions `# TODO` or `# FIXME`